internal/hud/tabview.go

```diff
@@ -1,26 +1,93 @@
 package hud
 
 import (
+	"fmt"
+
+	"github.com/gdamore/tcell"
+
 	"github.com/windmilleng/tilt/internal/hud/view"
 	"github.com/windmilleng/tilt/internal/rty"
 )
 
 type TabView struct {
 	view      view.View
 	viewState view.ViewState
+	tabState  view.TabState
 }
 
 func NewTabView(v view.View, vState view.ViewState) *TabView {
 	return &TabView{
 		view:      v,
 		viewState: vState,
+		tabState:  vState.TabState,
 	}
 }
 
 func (v *TabView) Build() rty.Component {
-	l := rty.NewConcatLayout(rty.DirHor)
+	l := rty.NewConcatLayout(rty.DirVert)
+	l.Add(v.buildTabs())
+
 	log := rty.NewTextScrollLayout("log")
-	log.Add(rty.TextString(v.view.Log.Tail(mainLogLineCount).String()))
+	log.Add(rty.TextString(v.log()))
 	l.Add(log)
+
 	return l
 }
+
+func (v *TabView) log() string {
+	var ret string
+	switch v.tabState {
+	case view.TabAllLog:
+		ret = v.view.Log.String()
```

Comment:
can you Tail(mainLogLineCount) these, to avoid the re-introducing the perf problems?